lib/ssl/src/tls_connection.erl

```diff
@@ -713,27 +733,41 @@ hello(internal, #client_hello{client_version = ClientVersion} = Hello,
             end
 
     end;
-hello(internal, #server_hello{} = Hello,      
+hello(internal, #server_hello{extensions = Extensions} = Hello,
       #state{connection_states = ConnectionStates0,
              connection_env = #connection_env{negotiated_version = ReqVersion} = CEnv,
 	     static_env = #static_env{role = client},
-             handshake_env = #handshake_env{renegotiation = {Renegotiation, _}},
+             handshake_env = #handshake_env{
+                 renegotiation = {Renegotiation, _},
+                 ocsp_stapling_state = OcspState} = HsEnv,
 	     ssl_options = SslOptions} = State) ->
+    %% check if server has sent a certifucate status message in hello
+    IsCertStatusRequested = maps:is_key(ocsp_stapling, Extensions),
     case tls_handshake:hello(Hello, SslOptions, ConnectionStates0, Renegotiation) of
 	#alert{} = Alert -> %%TODO
 	    ssl_connection:handle_own_alert(Alert, ReqVersion, hello,
                                             State#state{connection_env =
-                                                            CEnv#connection_env{negotiated_version = ReqVersion}});
+                                                            CEnv#connection_env{negotiated_version = ReqVersion},
+                                                        handshake_env = HsEnv#handshake_env{
+                                                            ocsp_stapling_state = OcspState#{
+                                                                is_server_hello_requested := IsCertStatusRequested}}
+                                                        });
         %% Legacy TLS 1.2 and older
 	{Version, NewId, ConnectionStates, ProtoExt, Protocol} ->
 	    ssl_connection:handle_session(Hello, 
-					  Version, NewId, ConnectionStates, ProtoExt, Protocol, State);
+					  Version, NewId, ConnectionStates, ProtoExt, Protocol,
+                      State#state{handshake_env = HsEnv#handshake_env{
+                          ocsp_stapling_state = OcspState#{
+                              is_server_hello_requested := IsCertStatusRequested}}});
         %% TLS 1.3
         {next_state, wait_sh, SelectedVersion} ->
             %% Continue in TLS 1.3 'wait_sh' state
             {next_state, wait_sh,
              State#state{
-               connection_env = CEnv#connection_env{negotiated_version = SelectedVersion}},
+               connection_env = CEnv#connection_env{negotiated_version = SelectedVersion},
+               handshake_env = HsEnv#handshake_env{
+                   ocsp_stapling_state = OcspState#{
+                       is_server_hello_requested := IsCertStatusRequested}}},
```

Comment:
Same comment about the single variable that represent the negotiated ocsp state.